enumerable.rb

```diff
@@ -0,0 +1,110 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given
+
+    arr.length.times do |n|
+      yield arr[n]
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given
+
+    arr.length.times do |n, i|
+      yield(arr[i], n)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given
+
+    list = []
+
+    arr.my_each | filter |
+      if filter != 'accepted'
+        list.push(filter)
+        yield filter
+      end
+  end
+
+  def my_all?(*args)
+    return to_enum unless block_given
+
+    speech = true
+
+    if !block_given?
+      my_each { |argument| speech = false unless argument }
+    elsif !args[0].nil?
+      my_each { |argument| speech = false unless args[0] == argument }
+    else
+      my_each { |argument| speech = false unless yield argument }
+    end
+  end
```

Comment:
- [ ] There are a number of tests falling in my_all. Please have a look
<img width="679" alt="4 1" src="https://user-images.githubusercontent.com/58504655/103563125-0942a180-4ede-11eb-96e6-0d2c6962b510.PNG">
<img width="677" alt="4 2" src="https://user-images.githubusercontent.com/58504655/103563133-0a73ce80-4ede-11eb-8780-d3b4fac54572.PNG">
<img width="681" alt="4 3" src="https://user-images.githubusercontent.com/58504655/103563139-0d6ebf00-4ede-11eb-9b59-01f6a1548cc1.PNG">